### PR TITLE
User in activity table

### PR DESCRIPTION
**Pull request by philipep-galdino**
# User field Added to Activity table

_Now we have an user field that needs to be filled/registered every time a new activity goes in and so you can get it on a basic Activity query. Also now we have a query to filter Activity by User ID too._

## How should this be tested?

1. Run the API
2. Create a new Wiki 
3. Run graphQL Playground and run the queries for User in Activities or Activities where User ID equals your User ID.

## Notes or observations

_I added the new query thinking it might be cool if we'd have a future feature where user wants to check out his activity history or something like that. There goes our new query running along with our new field added to the schema:_
image

## Linked issues

closes #00, closes #001, closes https://github.com/EveripediaNetwork/issues/issues/270

### CODE REVIEW

**Review by AshakaE — APPROVED** [review]
Looks good 👍

File: src/Database/Entities/activity.entity.ts [hunk 1]
```diff
@@ -29,6 +33,10 @@ class Activity {
   @PrimaryGeneratedColumn('uuid')
   id!: string
 
+  @Field(() => User)
+  @ManyToOne('User', 'user', { lazy: true })
+  user!: User
```
Comment (kesar) [inline src/Database/Entities/activity.entity.ts:38]: I think you will need to fix somehow the current data on the dev api in order to make this mandatory. Probably you need to run some small script / manually add the userId values from the old activities

File: src/App/activity.resolver.ts [hunk 2]
```diff
@@ -45,6 +51,22 @@ class ActivityResolver {
...
+  async activitiesByUser(@Args() args: ActivityArgsByUser) {
+    const repository = this.connection.getRepository(Activity)
+
+    return repository.find({
+      where: {
+        user: {
+          id: args.userId,
+        },
+      },
+      order: {
```
Comment (kesar) [inline src/App/activity.resolver.ts:65]: limit / offset are missing. you are extending PaginationArgs but not using those args

**Review by kesar — CHANGES_REQUESTED** [review]
limit / offset are missing. you are extending PaginationArgs but not using those args

File: src/App/activity.resolver.ts (hunk 2 above)
Comment (philipep-galdino) [reply src/App/activity.resolver.ts:67]: alright, done! forgot about that 😅

**Comment by philipep-galdino** [comment]
@AshakaE came up with a good SQL script to add the userId values from the old activities, once I get the ok here we'll merge and run the script on the db as we need the column there

**Comment by AshakaE** [comment]
> @AshakaE came up with a good SQL script to add the userId values from the old activities, once I get the ok here we'll merge and run the script on the db as we need the column there

image

**Review by kesar — APPROVED** [review]

**Pull request merged**